runner/utils/driver/docker/docker.go

```diff
@@ -0,0 +1,252 @@
+			Labels:       service.Labels,
+		}, &container.HostConfig{
+			// receiving memory in mega bytes converting into bytes
+			// convert received mill cpus to cpus by diving by 1000 then multiply with 100000 to get cpu quota
+			Resources: container.Resources{Memory: task.Resources.Memory * 1024 * 1024, CPUQuota: task.Resources.CPU * 100},
+		}, nil, containerName)
+		if err != nil {
+			logrus.Errorf("error applying service in docker unable to create container %s got error message - %v", containerName, err)
+			return err
+		}
+
+		if err := d.client.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+			logrus.Errorf("error applying service in docker unable to start container %s got error message - %v", containerName, err)
+			return err
+		}
+
+		// get ip address of service & store it in host file
+		data, err := d.client.ContainerInspect(ctx, resp.ID)
+		if err != nil {
+			logrus.Errorf("error applying service in docker unable to inspect container %s got error message  -%v", containerName, err)
+			return err
+		}
+		hostFile.AddHost(data.NetworkSettings.IPAddress, fmt.Sprintf("%s.%s-%s.svc.cluster.local", service.ID, service.ProjectID, task.ID))
+	}
+
+	// TODO CHECK IF THE BELOW FUNCTION IS NEEDED WHILE TESTING
+	// if err := hostFile.Save(); err != nil {
+	// 	logrus.Errorf("error applying service in docker unable to save host file - %v", err)
+	// 	return err
+	// }
+	return nil
+}
+
+// DeleteService removes every docker container related to specified service id
+func (d *docker) DeleteService(serviceId string) error {
+	ctx := context.Background()
+	containers, err := d.client.ContainerList(ctx, types.ContainerListOptions{})
+	if err != nil {
+		logrus.Errorf("error deleting service in docker unable to list containers got error message - %v", err)
+		return err
+	}
+
+	for _, containerInfo := range containers {
+		// there will be only 1 container name set
+		if len(containerInfo.Names) != 1 {
+			logrus.Errorf("error deleting service in docker containers length not equal to one")
+			return fmt.Errorf("error deleting service in docker containers length not equal to one")
+		}
+		// container name < serviceId-projectId-taskId-version >
+		containerName := containerInfo.Names[0]
+		if strings.Split(containerName, "-")[0] == serviceId {
+			// stop the container forcefully if status is running
+			if containerInfo.Status == "running" { // todo check the status
+				if err := d.client.ContainerKill(ctx, containerName, "SIGKILL"); err != nil {
+					logrus.Errorf("error deleting service in docker unable to kill container %s got error message - %v", containerName, err)
+					return err
+				}
+			}
+
+			// remove the container from host machine
+			if err := d.client.ContainerRemove(ctx, containerName, types.ContainerRemoveOptions{}); err != nil {
+				logrus.Errorf("error deleting service in docker unable to remove container %s got error message - %v", containerName, err)
+				return err
+			}
+		}
```

Comment:
There is a better way to remove.
```golang
        // Remove all the containers. This is save since we are filtering containers which belong to this allocation only.
	for _, c := range containers {
		_ = m.docker.ContainerRemove(ctx, c.ID, types.ContainerRemoveOptions{Force: true})
	}

```